scripts/circleci/bazel/setup-remote-execution.sh

```diff
@@ -16,4 +16,11 @@ openssl aes-256-cbc -d -in .circleci/gcp_token -md md5 -k ${GCP_DECRYPT_TOKEN} \
 echo "export GOOGLE_APPLICATION_CREDENTIALS=/home/.gcp_credentials" >> $BASH_ENV
 
 # Update the CircleCI Bazel configuration to always use remote execution.
-echo 'build --config=remote' >> .circleci/bazel.rc
+echo "build --config=remote" >> .circleci/bazel.rc
+
+# Only upload locally built results to the cache if we are running already commited code.
+if [[ -n "${CIRCLE_PR_NUMBER}" ]]; then
+  echo "build --remote_upload_local_results=false" >> .circleci/bazel.rc
```

Comment:
Are there any security implications since people could technically just modify this script in PRs and set it to `true`?

Reply:
The thinking is that if we see that happen and we are worried we can just clear out our cache if we want to consider it poisoned.

But the risk of a poisoned cache is really just that we could get tests that should fail show up as passing.  But we actually already hedge against that since we also test the changes against g3.

If we have concerns in the future we could come up with some other mechanism for locking things down, though I don't see that being necessary.

Reply:
> The thinking is that if we see that happen and we are worried we can just clear out our cache if we want to consider it poisoned.

That is a fair point but not always true because technically people could do this locally without even creating a PR. They have access to the GCP instance since the credentials are public in order to support RBE for PRs.

Though I'm not too worried about this either because I agree that Google3 can help and additionally we'll see if something is poisoned.

Last point I have that is not necessarily in scope for this PR. Since we already upload the local results like that and have the risk of a poisoned cache, can't we also just enable the RBE remote accept caching flag? We had this one disabled due to the risk of poisoned cache?

Reply:
To use RBE (and enter anything into the cache) require a credential (the service account key from our GCP project).  It is stored either locally in ones dev environment or in the circleCI environment. 

From `setup-remote-execution.sh`
```
echo "export GOOGLE_APPLICATION_CREDENTIALS=/home/.gcp_credentials" >> $BASH_ENV
```

I flipped the `remote_accept_cached` to `true` in `.circleci/bazel.rc`

Reply:
Yeah that's why we should be good turning on `remote_accept_cached` as well. We already live with the risk of poisoned cache and therefore setting `remote_accept_cached` should be fine as well :+1: